From 5bd0314bbe72ffab90c310d110986fc71165f121 Mon Sep 17 00:00:00 2001
From: Eli Collins <eli@cloudera.com>
Date: Mon, 27 Jun 2011 22:40:57 -0700
Subject: [PATCH 0995/1020] HDFS-2117. DiskChecker#mkdirsWithExistsAndPermissionCheck may return
 true even when the dir is not created.

Reason: Bug
Author: Eli Collins
Ref: CDH-3064
---
 src/core/org/apache/hadoop/util/DiskChecker.java |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/core/org/apache/hadoop/util/DiskChecker.java b/src/core/org/apache/hadoop/util/DiskChecker.java
index dde32d0..9f08946 100644
--- a/src/core/org/apache/hadoop/util/DiskChecker.java
+++ b/src/core/org/apache/hadoop/util/DiskChecker.java
@@ -129,6 +129,8 @@ public class DiskChecker {
       if (created) {
         localFS.setPermission(dir, expected);
         return true;
+      } else {
+        return false;
       }
     }
 
-- 
1.7.0.4

